tensorflow_datasets/audio/commonvoice.py

```diff
@@ -0,0 +1,126 @@
+            raise ValueError(
+                "language must be one of %s" %
+                _DOWNLOAD_LINKS.keys())
+        self._language = language
+        name = kwargs.get("name", None)
+        name = "%s_%s" % ("common_voice" if name is None else name, language)
+        kwargs["name"] = name
+        description = kwargs.get("description", None)
+        description = "Language Code: %s" % language if description is None else description
+        kwargs["description"] = description
+        super(CommonVoiceConfig, self).__init__(**kwargs)
+
+    @property
+    def download_urls(self):
+        """
+            Property returning Download URL based on Language Specified
+        """
+        return _DOWNLOAD_LINKS[self._language]
+
+
+def _generate_builder_configs():
+    """
+        Generates Builder Configs
+    """
+    configs = []
+    version = "1.0.0"
+    for k in _DOWNLOAD_LINKS.keys():
+        config = CommonVoiceConfig(version=version, language=k)
+        configs.append(config)
+    return configs
+
+
+class CommonVoice(tfds.core.GeneratorBasedBuilder):
+    """
+    Mozilla Common Voice Dataset
+    """
+    BUILDER_CONFIGS = _generate_builder_configs()
+    VERSION = tfds.core.Version("1.0.0")
+
+    def _info(self):
+        return tfds.core.DatasetInfo(
+            description=("Mozilla Common Voice Dataset"),
+            builder=self,
+            features=tfds.features.FeaturesDict({
+                "client_id": tfds.features.Text(),
+                "upvotes": tf.int32,
+                "downvotes": tf.int32,
+                "age": tfds.features.Text(),
+                "gender": tfds.features.Text(),
+                "accent": tfds.features.Text(),
+                "sentence": tfds.features.Text(),
+                "voice": tfds.features.Audio()
+            }),
+            urls=[u"https://voice.mozilla.org/en/datasets"]
+        )
+
+    def _split_generators(self, dl_manager):
+        dl_path = dl_manager.extract(
+            dl_manager.download(
+                self.builder_config.download_urls))
+        # Need to Check for replacement
+        clip_folder = os.path.join(dl_path, "clips")
+        return [tfds.core.SplitGenerator(
+            name=k,
+            num_shards=40,
+            gen_kwargs={
+                "audio_path": clip_folder,
+                "label_path": os.path.join(dl_path, "%s.tsv" % v)
+            }) for k, v in _SPLITS.items()]
+
+    def _generate_examples(self, audio_path, label_path):
+        """
+            Generate Voice Samples and Statements Given the Path to the Shared Audio Folder
+            and Path to the Train/Test/Validation File (.tsv)
+        """
+        with tf.io.gfile.GFile(label_path) as file_:
+            dataset = csv.DictReader(file_, delimiter="\t")
+            for row in dataset:
+                yield {
+                    "client_id": row["client_id"],
+                    "voice": os.path.join(audio_path, "%s.mp3" % row["path"]),
```

Comment:
Please add this dataset to setup.py to have pydub installed, similar to librispeech:
https://github.com/tensorflow/datasets/blob/96fe8c3156b43eb776456522ac739a0a6cdb86d2/setup.py#L75

Reply:
Done